src/Compilers/Core/Portable/Compilation/IOperation.cs

```diff
@@ -14,6 +14,12 @@ public interface IOperation
         /// Identifies the kind of the operation.
         /// </summary>
         OperationKind Kind { get; }
+
+        /// <summary>
+        /// Indicates whether the compiler has reported semantic errors about the operation.
+        /// </summary>
+        bool HasErrors { get; }
```

Comment:
Can you be more specific?  Does this mean an error directly at this operation level, or an error anywhere underneath? 

And, if this is the case, can we possibly have the errors also surfaced some way?

Also, if htis is only about Semantic errors, then maybe "HasSemanticErrors" would be better.  I would have had no idea this didn't include syntax based on hte name.